src/lib/storage/table.hpp

```diff
@@ -68,10 +69,13 @@ class Table : private Noncopyable {
 
  protected:
   std::vector<std::shared_ptr<Chunk>> _chunks;
+  std::recursive_mutex _chunk_access_lock;
   ChunkOffset _target_chunk_size;
   std::vector<std::string> _column_names;
   std::vector<std::string> _column_types;
   std::vector<bool> _column_nullables;
+  void compress_segment_and_add_to_chunk(ColumnID index, const std::shared_ptr<Chunk>& new_chunk,
```

Comment:
Please prepend protected members with `_`:
```suggestion
  void _compress_segment_and_add_to_chunk(ColumnID index, const std::shared_ptr<Chunk>& new_chunk,
```